src/Stan/Analysis/Analyser.hs

```diff
@@ -73,7 +73,9 @@ analyseNameMeta insId nameMeta patType hie@HieFile{..} =
             -- matches with the given nameMeta
             $ hieMatchNameMeta nameMeta hieId
             -- compatible with the given pattern
-            && any (hieMatchPatternType hie_types patType) typeIxs
+            && case typeIxs of
+                []    -> False
```

Comment:
Shouldn't it be True?

Reply:
I tried to match behaviour of the `any` function: it returns `False` when the list is empty. Maybe it makes sense in our case as well, since if there are no types to match against, we can't match and return `True`... :thinking: